topic-operator/src/main/java/io/strimzi/operator/topic/v2/ReconcilableTopic.java

```diff
@@ -4,16 +4,74 @@
  */
 package io.strimzi.operator.topic.v2;
 
+import io.micrometer.core.instrument.Timer;
 import io.strimzi.api.kafka.model.KafkaTopic;
 import io.strimzi.operator.common.Reconciliation;
+import io.strimzi.operator.common.metrics.MetricsHolder;
 
 /**
  * A topic to be reconciled
- * @param reconciliation The reconciliation
- * @param kt The topic
- * @param topicName The name of the topic in Kafka (spec.topicName, or metadata.name)
  */
-record ReconcilableTopic(Reconciliation reconciliation, KafkaTopic kt, String topicName) {
+public class ReconcilableTopic {
+    private Reconciliation reconciliation;
+    private KafkaTopic kt;
+    private String topicName;
+    private Timer.Sample sample;
+
+    /**
+     * @param reconciliation The reconciliation
+     * @param kt The topic
+     * @param topicName The name of the topic in Kafka (spec.topicName, or metadata.name)
+     */
+    public ReconcilableTopic(Reconciliation reconciliation, KafkaTopic kt, String topicName) {
+        this.reconciliation = reconciliation;
+        this.kt = kt;
+        this.topicName = topicName;
+    }
+
+    /**
+     * Returns the reconciliation.
+     * @return Reconciliation.
+     */
+    public Reconciliation reconciliation() {
+        return reconciliation;
+    }
+
+    /**
+     * Returns the Kafka topic.
+     * @return Kafka topic.
+     */
+    public KafkaTopic kt() {
+        return kt;
+    }
+
+    /**
+     * Returns the topic name.
+     * @return Topic name
+     */
+    public String topicName() {
+        return topicName;
+    }
+
+    /**
+     * Start the reconciliation timer.
+     * @param metrics Metrics holder
+     */
+    public void startTimer(MetricsHolder metrics) {
+        if (sample == null) {
+            sample = Timer.start(metrics.metricsProvider().meterRegistry());
+        }
+    }
+
+    /**
+     * Stop the reconciliation timer if present.
+     * @param metrics Metrics holder
+     */
+    public void stopTimer(MetricsHolder metrics) {
+        if (sample != null) {
```

Comment:
I find this pattern a little odd. I think the RT should be dumb holder of information. The interaction with the `MetricsHolder` could be done in the BTC, with the RT just holding the sample. That way it doesn't need to know about `MetricsHolder`, nor which metric is involved.

Reply:
I agree, will change that. Thanks.